src/Xrefcheck/Verify.hs

```diff
@@ -81,230 +87,324 @@ toVerifyRes = VerifyResult . either one (\() -> [])
-    | ExternalResourceInvalidUri
-    | ExternalResourceUnknownProtocol
-    | ExternalResourceUnavailable Status
-    | ExternalResourceSomeError Text
-    deriving (Show)
+  = FileDoesNotExist FilePath
+  | AnchorDoesNotExist Text [Anchor]
+  | AmbiguousAnchorRef FilePath Text (NonEmpty Anchor)
+  | ExternalResourceInvalidUri
+  | ExternalResourceUnknownProtocol
+  | ExternalHttpResourceUnavailable Status
+  | ExternalFtpResourceUnavailable FTPResponse
+  | ExternalFtpException FTPException
+  | ExternalResourceSomeError Text
+  deriving (Show)
 
 instance Buildable VerifyError where
-    build = \case
-        FileDoesNotExist file ->
-            "⛀  File does not exist:\n   " +| file |+ "\n"
-        AnchorDoesNotExist anchor similar ->
-            "⛀  Anchor '" +| anchor |+ "' is not present" +|
-            anchorHints similar
-        AmbiguousAnchorRef file anchor fileAnchors ->
-            "⛀  Ambiguous reference to anchor '" +| anchor |+ "'\n   " +|
-            "In file " +| file |+ "\n   " +|
-            "Similar anchors are:\n" +|
-                blockListF' "    -" build fileAnchors |+ "" +|
-            "   Use of such anchors is discouraged because referenced object\n\
-            \   can change silently whereas the document containing it evolves.\n"
-        ExternalResourceInvalidUri ->
-            "⛂  Invalid url\n"
-        ExternalResourceUnknownProtocol ->
-            "⛂  Bad url (expected 'http' or 'https')\n"
-        ExternalResourceUnavailable status ->
-            "⛂  Resource unavailable (" +| statusCode status |+ " " +|
-            decodeUtf8 @Text (statusMessage status) |+ ")\n"
-        ExternalResourceSomeError err ->
-            "⛂  " +| build err |+ "\n\n"
-      where
-        anchorHints = \case
-            []  -> "\n"
-            [h] -> ",\n   did you mean " +| h |+ "?\n"
-            hs  -> ", did you mean:\n" +| blockListF' "    -" build hs
+  build = \case
+    FileDoesNotExist file ->
+      "⛀  File does not exist:\n   " +| file |+ "\n"
+
+    AnchorDoesNotExist anchor similar ->
+      "⛀  Anchor '" +| anchor |+ "' is not present" +|
+      anchorHints similar
+
+    AmbiguousAnchorRef file anchor fileAnchors ->
+      "⛀  Ambiguous reference to anchor '" +| anchor |+ "'\n   " +|
+      "In file " +| file |+ "\n   " +|
+      "Similar anchors are:\n" +|
+          blockListF' "    -" build fileAnchors |+ "" +|
+      "   Use of such anchors is discouraged because referenced object\n\
+      \   can change silently whereas the document containing it evolves.\n"
+
+    ExternalResourceInvalidUri ->
+      "⛂  Invalid url\n"
+
+    ExternalResourceUnknownProtocol ->
+      "⛂  Bad url (expected 'http','https' or 'ftp')\n"
+
+    ExternalHttpResourceUnavailable status ->
+      "⛂  Resource unavailable (" +| statusCode status |+ " " +|
+      decodeUtf8 @Text (statusMessage status) |+ ")\n"
+
+    ExternalFtpResourceUnavailable status ->
+      "⛂  Resource unavailable (" +| status |+ ")\n"
```

Comment:
```suggestion
    ExternalFtpResourceUnavailable response ->
      "⛂  Resource unavailable (" +| response |+ ")\n"
```

Reply:
Also, the `response` might span multiple lines, so it's probably best to display it below `⛂  Resource unavailable`, instead of in between `(` `)`

Reply:
Discussed with Kirill, that this row exception can be a few words or few gigabytes